code/modules/antagonists/heretic/magic/rust_charge.dm

```diff
@@ -0,0 +1,51 @@
+// Rust charge, added so i could put the lion hunter rifle earlier up the tree
+/datum/action/cooldown/mob_cooldown/charge/rust
+	name = "rust charge"
+	desc = "A charge that must be started on a rusted tile and will destroy any rusted objects you come into contact with, will deal high damage to others and rust around you during the charge."
+	charge_distance = 10
+	charge_damage = 50
+	cooldown_time = 45 SECONDS
+
+/datum/action/cooldown/mob_cooldown/charge/rust/Activate(atom/target_atom)
+	if(HAS_TRAIT(get_step(owner, 0), TRAIT_RUSTY))
```

Comment:
```suggestion
	if(HAS_TRAIT(get_turf(owner), TRAIT_RUSTY))
```

Reply:
This may throw compile errors, honestly it's preferable if you separate the check out.
```dm
var/turf/open/start_turf = get_turf(owner)
if(istype(start_turf) && HAS_TRAIT(start_turf, TRAIT_RUSTY))
```
But also early return.